es-ds-components/components/es-rating.vue

```diff
@@ -0,0 +1,121 @@
+
+<template>
+    <rating
+        :modelValue="roundedRating"
+        :cancel="false" 
+        :readonly="readOnly"
+        @update:modelValue="update"
+        v-bind="$attrs"
+        data-testid="rating-test"
+        :pt="{
+            root: (options)  => ({
+                class: [
+                    'bg-transparent rounded-0 text-orange rating',
+                    {
+                        'reactive':  !options.props.readonly
+                    }
+                ]
+            }),   	
+            item: (options)  => ({
+                class:[
+                    { 
+                       
+                        'reactiveStar':  !options.props.readonly
+                    }
+                ],
+            })
+        }"
+    >
+    <template #officon>
+        <icon-star-empty
+            :width="width"
+            :height="height" />
+    </template>
+    <template #onicon>
+        <icon-star-full
+            :width="width"
+            :height="height" />
+    </template>
+    </rating>
+</template>
+
+<script setup lang="ts">
+import Rating from 'primevue/rating';
+const props = defineProps({
+    /**
+     * Starting rating
+     * 0-5; Avg will show half icons rounded down
+     */
+    rating: {
+        type: Number,
+        default: 0,
+        validator: (num) => num >= 0 && num <= 5,
+    },
+    /**
+     * Disable changing the rating
+     */
+    readOnly: {
+        type: Boolean,
+        default: true,
+    },
+    /**
+     * Icon Width
+     */
+    width: {
+        type: String,
+        default: '20px',
+        required: false,
+    },
+    /**
+     * Icon Height
+     */
+    height: {
+        type: String,
+        default: '20px',
+        required: false,
+    },
+    /**
+     * Round rating to nearest .5
+     */
+    rounded: {
+        type: Boolean,
+        default: true,
+        required: false,
+    },
+});
+
+var localRating = ref(props.rating)
```

Comment:
probably OK until we have linting, but most linters tend to suggest using block-scoped variables i.e. `const` when possible.